### PR TITLE
apps/examples: add uart_loopback_test

### PR DESCRIPTION
**Pull request by ewoodev**
This is a test example where you can short-circuit the uart tx and rx boards to perform a loopback test. This test sends "1234567890abcdefghijklmnopqrstuvwxyz" and recive same string

You can use CONIFG_UART_LOOPBACK_PORT to specify the port for the loopback test.

[TEST RESULT]
TASH>>uart_loopback
TASH>>######################### UART HW test START #########################
SEND SIZE:   37
SEND UART:   1234567890abcdefghijklmnopqrstuvwxyz
RECIVE SIZE: 37
RECIVE UART: 1234567890abcdefghijklmnopqrstuvwxyz
UART LOOPBACK TEST: PASSED
;######################### UART HW test END ###########################

### CODE REVIEW

File: apps/examples/uart_loopback_test/uart_loopback_test.c [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			close(fd);
+			return -1;
+		}
+#endif
+		ret_size = read(fd, (void *)read_ptr, remain_size);
+		remain_size -= ret_size;
+		read_ptr += ret_size;
+		read_total_size += ret_size;
+	}
+	printf("RECIVE SIZE: %d\n", read_total_size);
```
Comment (edwakuwaku) [inline apps/examples/uart_loopback_test/uart_loopback_test.c:112]: RECIVE -> RECEIVE

File: apps/examples/uart_loopback_test/Makefile [hunk 2]
```diff
@@ -0,0 +1,124 @@
+###########################################################################
+#
+# Copyright 2022 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/uart_loopback_test/Makefile:3]: 24

**Comment by sunghan-chang** [comment]
@ewoodev Any plan to update?

File: apps/examples/uart_loopback_test/uart_loopback_test.c (hunk 1 above)
Comment (ewoodev) [reply apps/examples/uart_loopback_test/uart_loopback_test.c]: Done

File: apps/examples/uart_loopback_test/Makefile (hunk 2 above)
Comment (ewoodev) [reply apps/examples/uart_loopback_test/Makefile]: done

File: apps/examples/uart_loopback_test/uart_loopback_test.c [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+/****************************************************************************
+ * Public functions
+ ****************************************************************************/
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int uart_loopback_test_main(int argc, char **argv)
+#endif
+{
+	pthread_t tx_tid, rx_tid;
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply apps/examples/uart_loopback_test/uart_loopback_test.c]: done

File: apps/examples/uart_loopback_test/uart_loopback_test.c [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	if (pthread_create(&tx_tid, NULL, (pthread_startroutine_t)uart_tx_loop, NULL) < 0) {
+		printf("Failed to create tx pthread for running logs\n");
+		pthread_cancel(rx_tid);
+		return -1;
+	}
+
+	pthread_join(tx_tid, &retval);
+	if (retval < 0) {
+		printf("uart tx thread exit with error %d\n", retval);
+		pthread_cancel(rx_tid);
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply apps/examples/uart_loopback_test/uart_loopback_test.c]: i changed pthread_join to pthread_detach

File: apps/examples/uart_loopback_test/uart_loopback_test.c [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			return -1;
+		}
+#endif
+		ret_size = read(fd, (void *)read_ptr, remain_size);
+		remain_size -= ret_size;
+		read_ptr += ret_size;
+		read_total_size += ret_size;
+	}
+	printf("RECIVE SIZE: %d\n", read_total_size);
+	printf("RECIVE UART: %s\n", read_buf);
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply apps/examples/uart_loopback_test/uart_loopback_test.c]: I changed "RECEIVE(%d): %s\n"

File: apps/examples/uart_loopback_test/uart_loopback_test.c [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+#endif
+
+#define UART_DEV_PATH		"/dev/ttyS%d"
+
+#define UART_POLL_TIMEOUT_MS	10000
+
+#define TEST_STR		"1234567890abcdefghijklmnopqrstuvwxyz"
+#define TEST_STR_LEN	37
+
+static int uart_tx_loop(void)
```
[the remark this answers is not in the document]
Comment (ewoodev) [reply apps/examples/uart_loopback_test/uart_loopback_test.c:50]: first time, i have made one time test, but i changed to repeat test

**Review by sunghan-chang — APPROVED** [review]

**Review by sunghan-chang — DISMISSED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**